src/Model/Scope/CompoundCondition.php

```diff
@@ -47,13 +47,7 @@ public function __construct(array $nestedConditions = [], string $junction = sel
                 if (!is_array($nestedCondition)) {
                     $nestedCondition = [$nestedCondition];
                 }
-
-                // array of OR nested conditions
-                if (count($nestedCondition) === 1 && is_array(reset($nestedCondition))) {
-                    $condition = new self(reset($nestedCondition), self::OR);
-                } else {
-                    $condition = new Condition(...$nestedCondition);
-                }
+                $condition = new Condition(...$nestedCondition);
```

Comment:
I would use a on-liner here:
`$condition = new Condition(...(array) $nestedCondition);`

Reply:
I prefer to keep this is easy to understand nesting levels

The non covered/buggy cases come mostly from bad coding style ;-)

Reply:
Please ...

Reply:
thank you :), never